### ISSUE TITLE
lazy symbol binding failed error

### ISSUE THREAD

**Issue by victoriazheng** [oclint/oclint#149]
When I try to run `oclint-json-compilation-database` against a json file generated by xctool, I get a `lazy symbol binding failed` error. I'm using oclint 0.7, but I tried it using oclint 0.8 dev still the same result.
I searched online for a while but didn't find anyone who had this issue before.

Full error message: 

``` sh
dyld: lazy symbol binding failed: Symbol not found: __ZN6oclint7RuleSetC1EPNS_8RuleBaseE
  Referenced from: .../lib/oclint/rules/libAvoidBranchingStatementAsLastInLoopRule.dylib
  Expected in: flat namespace

dyld: Symbol not found: __ZN6oclint7RuleSetC1EPNS_8RuleBaseE
  Referenced from: .../lib/oclint/rules/libAvoidBranchingStatementAsLastInLoopRule.dylib
  Expected in: flat namespace
```
